File: client-ios/Sources/Network/ApiModels.swift
import Foundation

/// API models matching the Ambient Intelligence protocol.
///
/// See: protocol/SPECIFICATION.md for full protocol details

struct PublicKeyResponse: Codable {
    let publicKey: String
    
    enum CodingKeys: String, CodingKey {
        case publicKey = "public_key"
    }
}

struct SubmitJobRequest: Codable {
    let encryptedPrompt: String
    let clientPubkey: String
    
    enum CodingKeys: String, CodingKey {
        case encryptedPrompt = "encrypted_prompt"
        case clientPubkey = "client_pubkey"
    }
}

struct SubmitJobResponse: Codable {
    let jobId: String
    
    enum CodingKeys: String, CodingKey {
        case jobId = "job_id"
    }
}

struct JobStatusResponse: Codable {
    let status: String // "running", "complete", "failed"
    let encryptedResponse: String?
    let errorMessage: String?
    
    enum CodingKeys: String, CodingKey {
        case status
        case encryptedResponse = "encrypted_response"
        case errorMessage = "error_message"
    }
}

/// Coordinator endpoints for dynamic node discovery
struct CoordinatorNodesResponse: Codable {
    let nodes: [NodeInfo]
}

struct NodeInfo: Codable {
    let nodeId: String
    let address: String
    let publicKey: String
    let capabilities: [String]
    
    enum CodingKeys: String, CodingKey {
        case nodeId = "node_id"
        case address
        case publicKey = "public_key"
        case capabilities
    }
}

File: client-android/src/main/java/com/ambient/client/network/ApiModels.kt
package com.ambient.client.network

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * API models matching the Ambient Intelligence protocol.
 * 
 * See: protocol/SPECIFICATION.md for full protocol details
 */

@Serializable
data class PublicKeyResponse(
    @SerialName("public_key")
    val publicKey: String
)

@Serializable
data class SubmitJobRequest(
    @SerialName("encrypted_prompt")
    val encryptedPrompt: String,
    
    @SerialName("client_pubkey")
    val clientPubkey: String
)

@Serializable
data class SubmitJobResponse(
    @SerialName("job_id")
    val jobId: String
)

@Serializable
data class JobStatusResponse(
    val status: String, // "running", "complete", "failed"
    
    @SerialName("encrypted_response")
    val encryptedResponse: String? = null,
    
    @SerialName("error_message")
    val errorMessage: String? = null
)

/**
 * Coordinator endpoints for dynamic node discovery
 */
@Serializable
data class CoordinatorNodesResponse(
    val nodes: List<NodeInfo>
)

@Serializable
data class NodeInfo(
    @SerialName("node_id")
    val nodeId: String,
    
    val address: String,
    
    @SerialName("public_key")
    val publicKey: String,
    
    val capabilities: List<String> = emptyList()
)
